File: 73-set-matrix-zeroes/73-set-matrix-zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
    let zeroArr = [];
    
    matrix.forEach((val, row) => {
        val.forEach((val, col) => {
            if (val === 0) zeroArr.push([row, col]);
        });
    });
    for (const element of zeroArr) {
        matrix.forEach((val, idx, arr) => {
            val[element[1]] = 0
            if (idx === element[0]) {
                for (let i = 0,len = matrix[idx].length; i < len; ++i )
                    matrix[idx][i] = 0;
            }
        });
    }
    return zeroArr;
};

File: 73-set-matrix-zeroes/73-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        m = len(matrix)
        n = len(matrix[0])
        row = set()
        col = set()
        for i in range(0, m):
            for j in range(0, n):
                if (matrix[i][j] == 0):
                    row.add(i)
                    col.add(j)

        row = list(row)
        col = list(col)
        for i in row:
            for j in range(0, n):
                matrix[i][j] = 0		
        for j in col:
            for i in range(0, m):
                matrix[i][j]  = 0
        return matrix